doc/alertmanager-mixin/alerts.libsonnet

```diff
@@ -25,9 +25,11 @@
             expr: |||
               # Without max_over_time, failed scrapes could create false negatives, see
               # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
-                max_over_time(alertmanager_cluster_members{%(alertmanagerSelector)s}[5m])
+                max by (%(alertmanagerNameLabels)s) (max_over_time(alertmanager_cluster_members{%(alertmanagerSelector)s}[5m]))
               < on (%(alertmanagerClusterLabels)s) group_left
-                count by (%(alertmanagerClusterLabels)s) (max_over_time(alertmanager_cluster_members{%(alertmanagerSelector)s}[5m]))
+                count by (%(alertmanagerClusterLabels)s) (
+                    max by (%(alertmanagerNameLabels)s) (max_over_time(alertmanager_cluster_members{%(alertmanagerSelector)s}[5m]))
+                )
```

Comment:
I don't think this is a generally valid solution. If `alertmanagerNameLabels` are not a superset of `alertmanagerClusterLabels`, the `max by (%(alertmanagerNameLabels)s)` will aggregate away the labels we need for the matching `on (%(alertmanagerClusterLabels)s)`.

I think the issue you describe is real, but is it relevant? The overcounting will only happen for a short while, not much more than 5m. And that's presumably the reason why the `for` clause below is `10m`. Have you actually seen the alert firing falsely? Would it have helped to use an even longer `for` duration?

Reply:
Thanks for the review @beorn7, I see now that the additional `max` aggregation might be introducing a bug in the alert.

To your question on whether the alert fires in our case, the answer is yes, we see it happen. I've attached a screenshot counting the number of series coming from alertmanager. We have a cluster of 3 instances and the count between 1:42:00 and 1:53:12 is greater than 3 as seen in the screenshot.
![Screenshot from 2021-06-15 17-09-49](https://user-images.githubusercontent.com/1286231/122078928-3d12e300-cdfd-11eb-89d2-1f1c932f6b45.png)

A longer `for` clause would have helped in our case and is a viable solution. The downside is that such a change might affect the false negative rate, or prolong the time it takes to detect.

Reply:
We managed to track down the reason for the overcounting. In our case, alertmanager pods were getting evicted several times in a short period of time which produced multiple stale series. This is why the graph attached above shows a peak which lasts longer than 10 minutes.

In order to avoid complicating the alert expression, I can change the PR to increase the duration to 15m in order to reduce the chance of a false positive. What do you think about this solution?

Reply:
I think trying a longer `for` duration is probably the best. After all, those frequent evictions can actually lead to weird situations where different AM cluster members have different ideas about their peers, and that's what this alert is about. However, it's not an urgent alert. If those things happen, alerts should still reach their receivers "at least once" (which is the desired semantics). So in doubt, we could go even longer. But 15m sounds like a good start.

Reply:
That makes complete sense. I've extended the `for` duration to `15m`.